### PR TITLE
Implements viem

### PR DESCRIPTION
**Pull request by Emmanueldmlr**
# replace ethers and use viem on iq.wiki

_We need to replace ethers usage with viem.

fixes https://github.com/EveripediaNetwork/issues/issues/1158

### CODE REVIEW

File: src/config/wagmi.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { chain, configureChains, Connector } from 'wagmi'
 import { alchemyProvider } from 'wagmi/providers/alchemy'
 import { infuraProvider } from 'wagmi/providers/infura'
 import { publicProvider } from 'wagmi/providers/public'
-import { AlchemyProvider, Network } from '@ethersproject/providers'
+import { AlchemyProvider } from '@ethersproject/providers'
```
Comment (kesar) [inline src/config/wagmi.ts:7]: its still this needed?

**Comment by kesar** [comment]
please add in the PR the missing parts to solve in future iterations

File: src/config/wagmi.ts (hunk 1 above)
Comment (Emmanueldmlr) [reply src/config/wagmi.ts:7]: Still WIP.

**Comment by Emmanueldmlr** [comment]
Missing Part for Future Iteration:

- [ ]  Integrate getAvatar from Viem
- [ ]  Removes Ethers completely once Wagmi updates their package to use Viem.
- [ ]  Replace BigNumber Type from Ethers

**Comment by kesar** [comment]
please replace web3-token w our new signer ( https://github.com/EveripediaNetwork/aiq-ui/commit/33b0bfc57297f9c7f55b0dfe81284dc324d7185e ) also make sure it uses viem instead of ethers to sign

File: package.json [hunk 2]
```diff
@@ -37,6 +37,7 @@
     "@ethersproject/units": "^5.6.1",
     "@everipedia/iq-utils": "^0.1.4",
     "@everipedia/wagmi-magic-connector": "^0.7.0",
+    "@everipedia/web3-signer": "^0.0.2",
```
Comment (Royal-lobster) [inline package.json:40]: @Emmanueldmlr update this to latest i think its now 0.0.11

**Review by Royal-lobster — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -37,6 +37,7 @@
     "@ethersproject/units": "^5.6.1",
     "@everipedia/iq-utils": "^0.1.4",
     "@everipedia/wagmi-magic-connector": "^0.7.0",
+    "@everipedia/web3-signer": "^0.0.9",
```
Comment (kesar) [inline package.json:40]: i think its 11 latest one, does it work as expected? did you notice anything different?

Comment (Emmanueldmlr) [reply package.json:40]: 9 works as expected. At the time of the upgrade, It was 9. I will upgrade it to 11 once I switch back to this task.

**Comment by Emmanueldmlr** [comment]
image

@kesar @Royal-lobster . I think it's 9

**Comment by Royal-lobster** [comment]
> @kesar @Royal-lobster . I think it's 9

Ohh true, it is the latest one

**Comment by Royal-lobster** [comment]
> so you are not getting any error in console? because I think @Royal-lobster you did in iq-social right?

wierd, let me check again

**Comment by Royal-lobster** [comment]
Yeah, now i am not able to reproduce the errors again, probably was due to artifacts from yarn link previous 🤔.

**Comment by kesar** [comment]
nice I guess we need to replace web3-token in backend btw

**Comment by Emmanueldmlr** [comment]
> nice I guess we need to replace web3-token in backend btw

Yes actually.

**Comment by Royal-lobster** [comment]
> nice I guess we need to replace web3-token in backend btw

Yes ! i will create a pr for it

**Comment by kesar** [comment]
i guess we can merge this one and then replace ethers once we get new wagmi version

**Pull request merged**